Framework/LiveData/src/ISIS/ISISHistoDataListener.cpp

```diff
@@ -124,6 +124,15 @@ bool ISISHistoDataListener::connect(const Poco::Net::SocketAddress &address) {
 
   loadTimeRegimes();
 
+  // Create dummy workspace to store instrument data
+  bufferWorkspace = WorkspaceFactory::Instance().create("Workspace2D", 1, 1, 1);
```

Comment:
In addition to the instrument data `m_bufferWorkspace` stores some axis information. That is why I don't want to use `SetInstrument()`. I also don't want to use`clone()` instead of `create()` from parent because I can't find a regular method to set dimensions, and I need the `shared_ptr` for this create. But `WorkspaceCreation` utilities produces `uniqe_ptr`, but we need `shared_ptr`.